llvm/lib/Target/RISCV/RISCV.td

```diff
@@ -170,6 +170,24 @@ def HasStdExtZvamo : Predicate<"Subtarget->hasStdExtZvamo()">,
                               AssemblerPredicate<(all_of FeatureExtZvamo),
                               "'Zvamo'(Vector AMO Operations)">;
 
+// CORE-V extensions
+// These extensions can either be enabled individually with their extension
+// name or all together with `+xcorev`
+def FeatureExtCoreV
+    : SubtargetFeature<"xcorev", "HasStdExtCoreV", "true",
+                       "'XCoreV' (CORE-V extensions)">;
+def HasStdExtCoreV : Predicate<"Subtarget->hasStdExtCoreV()">,
+                              AssemblerPredicate<(all_of FeatureExtCoreV),
+                              "'XCoreV' (CORE-V extensions)">;
+// CORE-V Hardware Loop Extension
+def FeatureExtCoreVHwlp
+    : SubtargetFeature<"xcorevhwlp", "HasStdExtCoreVHwlp", "true",
+                       "'XCoreVHwlp' (Hardware Loop)">;
+def HasStdExtCoreVHwlp
+    : Predicate<"Subtarget->hasStdExtCoreVHwlp() || Subtarget->hasStdExtCoreV()">,
```

Comment:
I don't think you need this, I would make the Xcorev extension imply this one. See what we did for Bitmanip as an example.